frontend/src/container/NewDashboard/ComponentsSlider/index.tsx

```diff
@@ -1,63 +1,90 @@
 import { initialQueriesMap, PANEL_TYPES } from 'constants/queryBuilder';
 import { queryParamNamesMap } from 'constants/queryBuilderQueryNames';
+import { useUpdateDashboard } from 'hooks/dashboard/useUpdateDashboard';
 import { useIsDarkMode } from 'hooks/useDarkMode';
-import { useNotifications } from 'hooks/useNotifications';
 import history from 'lib/history';
+import { useDashboard } from 'providers/Dashboard/Dashboard';
 import { CSSProperties, useCallback } from 'react';
-import { connect, useSelector } from 'react-redux';
-import { bindActionCreators, Dispatch } from 'redux';
-import { ThunkDispatch } from 'redux-thunk';
-import {
-	ToggleAddWidget,
-	ToggleAddWidgetProps,
-} from 'store/actions/dashboard/toggleAddWidget';
-import { AppState } from 'store/reducers';
-import AppActions from 'types/actions';
-import DashboardReducer from 'types/reducer/dashboards';
+import { v4 as uuid } from 'uuid';
 
 import menuItems from './menuItems';
 import { Card, Container, Text } from './styles';
 
-function DashboardGraphSlider({ toggleAddWidget }: Props): JSX.Element {
-	const { dashboards } = useSelector<AppState, DashboardReducer>(
-		(state) => state.dashboards,
-	);
+function DashboardGraphSlider(): JSX.Element {
+	const isDarkMode = useIsDarkMode();
+
+	const { handleToggleDashboardSlider, layouts } = useDashboard();
 
-	const { notifications } = useNotifications();
+	const { selectedDashboard } = useDashboard();
 
-	const [selectedDashboard] = dashboards;
-	const { data } = selectedDashboard;
+	const { data } = selectedDashboard || {};
+
+	const updateDashboardMutation = useUpdateDashboard();
 
 	const onClickHandler = useCallback(
 		(name: PANEL_TYPES) => (): void => {
-			try {
-				const emptyLayout = data.layout?.find((e) => e.i === 'empty');
-
-				if (emptyLayout === undefined) {
-					notifications.error({
-						message: 'Please click on Add Panel Button',
-					});
-					return;
-				}
+			const id = uuid();
 
-				toggleAddWidget(false);
+			updateDashboardMutation.mutateAsync(
+				{
+					uuid: selectedDashboard?.uuid || '',
+					data: {
+						title: data?.title || '',
+						variables: data?.variables || {},
+						description: data?.description || '',
+						name: data?.name || '',
+						tags: data?.tags || [],
+						layout: [
+							{
+								i: id,
+								w: 6,
+								x: 0,
+								h: 2,
+								y: 0,
+							},
+							...(layouts.filter((layout) => layout.i !== PANEL_TYPES.EMPTY_WIDGET) ||
+								[]),
+						],
+						widgets: [
+							...(data?.widgets || []),
+							{
+								id,
+								title: '',
+								description: '',
+								isStacked: false,
+								nullZeroValues: '',
+								opacity: '',
+								panelTypes: name,
+								query: initialQueriesMap.metrics,
+								timePreferance: 'GLOBAL_TIME',
+							},
+						],
+					},
+				},
+				{
+					onSuccess: (data) => {
```

Comment:
Can we handle `onError` here as well?

Reply:
no clarity from product to handle

Reply:
though added default something went wrong